detox/android/detox/src/main/java/com/wix/detox/espresso/scroll/LinearSwiper.kt

```diff
@@ -0,0 +1,83 @@
+package com.wix.detox.espresso.scroll
+
+import android.view.MotionEvent
+import androidx.test.espresso.UiController
+import com.wix.detox.espresso.action.common.MotionEvents
+
+/**
+ * The delay between each motion event.
+ * Reducing this value may fail the swipe on different devices. Please change with caution.
+ */
+private const val EVENT_DELAY = 20L
+
+/**
+  * Implementation of @see DetoxSwiper that swipes in a linear fashion uses const delay between events.
+ */
+class LinearSwiper @JvmOverloads constructor(
```

Comment:
That's great and all (it really is!), but why the dup code? It's identical to `FlinglessSwiper` in practically everything but `calcEventTime()`, isn't it? I think there's room for sharing everything we can via a base class 🤔

Reply:
very good point